tests/logical_lattice_ops/logical_lattice_ops_test.py

```diff
@@ -63,3 +67,102 @@ def test_get_operating_patches(self):
+
+    c3 = PauliOpCircuit(1)
+    c3.add_pauli_block(PauliRotation.from_list([X], Fraction(-1, 4)))
+    c3.add_pauli_block(Measurement.from_list([Z]))
+    return [(c1, 2), (c2, 4), (c3, 1)]
+
+
+def generate_tests_count_magic_states():
+    with open("assets/demo_circuits/nontrivial_state.qasm") as input_file:
+        c1 = segmented_qasm_parser.parse_str(input_file.read())
+
+    with open("assets/demo_circuits/bell_pair.qasm") as input_file:
+        c2 = segmented_qasm_parser.parse_str(input_file.read())
+
+    return [(c1, 1), (c2, 0)]
+
+
+def generate_tests_circuit_to_single_patch_measurement():
+    m1 = Measurement.from_list([X], isNegative=False)
+    m2 = Measurement.from_list([I, Z], isNegative=False)
+
+    c1 = PauliOpCircuit(1)
+    c1.add_pauli_block(m1)
+
+    c2 = PauliOpCircuit(2)
+    c2.add_pauli_block(m2)
+
+    return [(c1, m1), (c2, m2)]
+
+
+def generate_tests_circuit_to_multi_body_measurements():
+    m1 = Measurement.from_list([Z, X], isNegative=True)
+    m2 = Measurement.from_list([Z, I, I, Z], isNegative=False)
+
+    c1 = PauliOpCircuit(2)
+    c1.add_pauli_block(m1)
+
+    c2 = PauliOpCircuit(4)
+    c2.add_pauli_block(m2)
+
+    return [(c1, m1), (c2, m2)]
+
+
+class TestLogicalLatticeComputation:
+    @pytest.mark.parametrize("input, expected", generate_tests_num_logical_qubits())
+    def test_num_logical_qubits(self, input, expected):
+        logical_computation = LogicalLatticeComputation(input)
+        assert logical_computation.num_logical_qubits() == expected
+
+    @pytest.mark.parametrize("input, expected", generate_tests_count_magic_states())
+    def test_count_magic_states(self, input, expected):
+        logical_computation = LogicalLatticeComputation(input)
+        assert logical_computation.count_magic_states() == expected
+
+    @pytest.mark.parametrize(
+        "circuit, measurement", generate_tests_circuit_to_single_patch_measurement()
+    )
+    def test_circuit_to_single_patch_measurement(
+        self, circuit: PauliOpCircuit, measurement: Measurement
+    ):
+        logical_comp = LogicalLatticeComputation(circuit)
+        patch_measurement = logical_comp.circuit_to_patch_measurement(logical_comp.circuit.ops[0])
+
+        assert isinstance(patch_measurement, SinglePatchMeasurement)
+        assert [patch_measurement.op] == list(measurement.get_ops_map().values())
+
+    @pytest.mark.parametrize(
+        "circuit, measurement", generate_tests_circuit_to_multi_body_measurements()
+    )
+    def test_circuit_to_multi_body_measurement(
+        self, circuit: PauliOpCircuit, measurement: Measurement
+    ):
+        logical_comp = LogicalLatticeComputation(circuit)
+        patch_measurement = logical_comp.circuit_to_patch_measurement(logical_comp.circuit.ops[0])
+        assert isinstance(patch_measurement, MultiBodyMeasurement)
+        assert list(patch_measurement.patch_pauli_operator_map.values()) == list(
+            measurement.get_ops_map().values()
```

Comment:
I'm surprised the order works here

Reply:
i think it's because both use the same logic. `get_ops_map()` populates as long as it is not identity. So does `circuit_to_patch_measurement` while populating the `patch_pauli_operator_map()`. Could this a be possible point of failure in the future? I think I wrote tests that are robust. Do you have additional test cases in mind for this particular test?

Reply:
I don't know ... usually when comparing maps you check the keys as sets and then compare key vale pairs or do some kind of sorting. But since this is a test we'll fix it only if it breaks